pkg/analyze/text_analyze.go

```diff
@@ -115,54 +115,66 @@ func analyzeRegexPattern(pattern string, collected []byte, outcomes []*troublesh
 		return nil, errors.Wrapf(err, "failed to compile regex: %s", pattern)
 	}
 
-	var failOutcome *troubleshootv1beta2.SingleOutcome
-	var passOutcome *troubleshootv1beta2.SingleOutcome
-	for _, outcome := range outcomes {
-		if outcome.Fail != nil {
-			failOutcome = outcome.Fail
-		} else if outcome.Pass != nil {
-			passOutcome = outcome.Pass
-		}
-	}
 	result := AnalyzeResult{
 		Title:   checkName,
 		IconKey: "kubernetes_text_analyze",
 		IconURI: "https://troubleshoot.sh/images/analyzer-icons/text-analyze.svg",
 	}
 
-	reMatch := re.MatchString(string(collected))
-	failWhen := false
-	if failOutcome != nil && failOutcome.When != "" {
-		failWhen, err = strconv.ParseBool(failOutcome.When)
-		if err != nil {
-			return nil, errors.Wrapf(err, "failed to process when statement: %s", failOutcome.When)
-		}
-	}
-	passWhen := true
-	if passOutcome != nil && passOutcome.When != "" {
-		passWhen, err = strconv.ParseBool(passOutcome.When)
-		if err != nil {
-			return nil, errors.Wrapf(err, "failed to process when statement: %s", passOutcome.When)
-		}
-	}
+	for _, outcome := range outcomes {
+		if outcome.Fail != nil {
+			isMatch := re.MatchString(string(collected))
```

Comment:
this can be called only once before the loop

Reply:
good catch! Fixed, Thx!